Automattic/jetpack
# Customizer widgets: enqueue script dependencies

Fixes https://github.com/Automattic/jetpack/issues/19980

Related to:
https://github.com/Automattic/jetpack/pull/20257


#### Changes proposed in this Pull Request

In https://github.com/Automattic/jetpack/pull/20044 we wrote the `view.js` script to the page so that blocks would have their scripts immediately available in the preview window after insertion from the **Customizer > Widgets** panel.

To ensure the inserted blocky wock has all the scripts/dependencies it needs to thrive we also jave to add its dependencies.

We've set a `defer` attribute to the block's `view.js` script to make sure it runs when the DOM is ready.

<img width="1069" alt="Screen Shot 2021-07-09 at 2 16 54 pm" src="https://user-images.githubusercontent.com/6458278/125024306-d9aa6700-e0c3-11eb-9c25-e8bdcc1c389c.png">

As a bonus, we've also removed the invalid `isLarge` prop on the `<FormFileUpload />` component in the Slideshow block. This is free. You don't have to pay for it. :) It was throwing the following error:

<img width="1205" alt="Screen Shot 2021-07-09 at 2 11 24 pm" src="https://user-images.githubusercontent.com/6458278/125024363-f3e44500-e0c3-11eb-8432-4691d59bf37d.png">

ALSO!

We've added a Twenty Twenty One compat fix to ensure the slideshow doesn't break out of the widget container:

**Before**
<img width="500" alt="Screen Shot 2021-07-15 at 8 09 47 am" src="https://user-images.githubusercontent.com/6458278/125699664-1033db80-a51e-4d1b-90aa-20cdf7136215.png">



**After**
<img width="500" alt="Screen Shot 2021-07-15 at 8 08 30 am" src="https://user-images.githubusercontent.com/6458278/125699669-133b8f88-b263-4d41-99cc-131b4cce5610.png">



#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

Running Jetpack + WordPress 5.8 or above, head over to **Customizer > Widgets**

Add the following blocks:

1. Tiled Gallery: add images and play with the gallery in the preview.
2. Slideshow: add images and play with the slideshow in the preview. Check that the console doesn't complain about the invalid `isLarge` prop.
3. Gif: search for GIF (try ponies!)
4. Other blocks (except Maps block 😄 ). Try to break it!
5. Switch to the 2021 theme and make sure the widgets (the slideshow in particular) look dandy

They should work in both the widgets editor and the preview (and the frontend).

Save and refresh the Customizer. It should look as you set it.
